api/src/main/java/io/strimzi/api/kafka/model/template/KafkaBridgeTemplate.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.template;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.api.kafka.model.UnknownPropertyPreserving;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Representation of a template for Kafka Bridge resources.
+ */
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({
+        "deployment", "pod", "apiService"})
+@EqualsAndHashCode
+public class KafkaBridgeTemplate implements Serializable, UnknownPropertyPreserving {
+    private static final long serialVersionUID = 1L;
+
+    private ResourceTemplate deployment;
+    private PodTemplate pod;
+    private ResourceTemplate apiService;
```

Comment:
Is `apiService` the right name here? It is an API of kind, but I'm not 100% sure. What do others think?

Reply:
@ppatierno @tombentley ^

Reply:
TBH I'd like to understand a little about whether _other_ services might be added in the future. Yesterday @ppatierno and I discussed how the consumer creation response includes a URL which includes the host. Potentially this would allow each replica in a Deployment of the bridge to return a URL to a host via a headless `Service` (similar to the Kafka bootstrap, metadata pattern). What that would mean for this comment is we'd potentially have a `bootstrapService` and a `bridgeService`.

Reply:
The headless service is usable only within the cluster. It would not work from outside.

Reply:
It's the pattern which is important. For exposing outside the cluster we'd end up having to use routes, etc just as we do for Kafka. But the point is that this pattern allows you to have Deployments with > 1 replica for consumers, because the client ends up communicating with the right bridge instance via the direct URL. That comes at a cost because communicating directly to that bridge instance is not HA, so the client application needs the error handling to recover from that. But it's a way of getting scalability. I'm also not sure it plays nicely with a reverse proxy sitting in front. So there are pros and cons, and we might ultimately decide we don't want to go this route.

Reply:
> But the point is that this pattern allows you to have Deployments with > 1 replica for consumers, because the client ends up communicating with the right bridge instance via the direct URL.

IMHO it doesn't unless you implement clustering between the instances to return the address of the right node. Because without it:

1) Client opens an consumer on replica 1 and gets back the replica address of this replica
2) It consumes the messages using the replica address
3) The pod restarts and looses the host information
4) The new instance connects to replica 2 and starts a new consumer with replica 2 address
5) The client scales and the new pod connects to replica 3 and gets replica 3 address etc.

This is much more then just sticky sessions. So I do not think this alone is a solution.